src/firefly/java/edu/caltech/ipac/firefly/server/catquery/GatorDD.java

```diff
@@ -39,7 +56,63 @@
                 @ParamDoc(name = CatalogRequest.SERVICE_ROOT, desc = "the part of the URL string that specifies the service and first params. " +
                         "optional: almost never used")
         })
-public class GatorDD extends BaseGator {
+public class GatorDD extends DbProcessor {
+
+    public FileInfo createDbFile(TableServerRequest treq) throws DataAccessException {
+        DbAdapter dbAdapter = DbAdapter.getAdapter(treq);
+        File dbFile = new File(ServerContext.getPermWorkDir(), "GatorDD." + dbAdapter.getName());
+        DataGroup dd = new DataGroup("DD for GatorDD", new DataType[]{
+                new DataType("name", String.class),
+                new DataType("description", String.class),
+                new DataType("units", String.class),
+                new DataType("indx", String.class),
+                new DataType("dbtype", String.class),
+                new DataType("tableflg", Integer.class),
+                new DataType("sel", String.class)
+        });
+        TableDbUtil.createDDTbl(dbFile, dd, dbAdapter);
+        TableDbUtil.setDbMetaInfo(treq, DbAdapter.getAdapter(treq), dbFile);
+        return new FileInfo(dbFile);
+    }
+
+    @Override
+    protected DataGroupPart getDataset(TableServerRequest treq, File dbFile) throws DataAccessException {
```

Comment:
`getDataset` seems like mostly baseclass/reusable code. `fetchDataInfoTable` appears to be the business logic. Should the code in `getDataset` be moved up into `DbProcessor`?

Reply:
Normally, you would create the db during createDbFile.  There after, you skip createDbFile and use getDataset to query it.  GatorDD is not typical.  The data is very static.  So, instead of having a new database for each catalog, I put them all into one database as it's being requested.  That's why the logic is a bit different.